File: app/frontend/src/types/todo.ts
export interface Todo {
  id: string
  github_user_id: string
  title: string
  description: string | null
  completed: boolean
  created_at: string
  updated_at: string
}

export interface TodoListResult {
  items: Todo[]
  total: number
  page: number
  limit: number
}

export interface TodoEventAudit {
  event_id: string
  schema_version: number
  event_type: string
  occurred_at: string
  processed_at: string
  environment: string
  github_user_id: string
  todo_id: string
  todo: Todo | null
}

export interface TodoEventAuditListResult {
  items: TodoEventAudit[]
  total: number
  page: number
  limit: number
}

export interface CreateTodoInput {
  title: string
  description?: string | null
}

export interface UpdateTodoInput {
  title?: string
  description?: string | null
  completed?: boolean
}

File: app/todo-service/internal/domain/todo.go
package domain

import (
	"errors"
	"time"

	"github.com/google/uuid"
)

var (
	ErrInvalidInput       = errors.New("invalid input")
	ErrUserNotFound       = errors.New("user not found")
	ErrTodoNotFound       = errors.New("todo not found")
	ErrProfileUnavailable = errors.New("profile service unavailable")
	ErrUnauthorized       = errors.New("unauthorized")
	ErrAuthUnavailable    = errors.New("authentication service unavailable")
)

type Todo struct {
	ID           uuid.UUID `json:"id"`
	GithubUserID uuid.UUID `json:"github_user_id"`
	Title        string    `json:"title"`
	Description  *string   `json:"description"`
	Completed    bool      `json:"completed"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

type ListResult struct {
	Items []Todo `json:"items"`
	Total int64  `json:"total"`
	Page  uint32 `json:"page"`
	Limit uint32 `json:"limit"`
}

type CreateInput struct {
	Title       string
	Description *string
}

type UpdateInput struct {
	Title          *string
	DescriptionSet bool
	Description    *string
	Completed      *bool
}
